src/packages/packages.controller.ts

```diff
@@ -0,0 +1,198 @@
+import { 
+  Controller, 
+  Get, 
+  Param, 
+  StreamableFile, 
+  NotFoundException, Res
+} from '@nestjs/common';
+import { EntityManager } from '@mikro-orm/postgresql';
+import { Package } from '../model/package.entity';
+import { Response } from 'express';
+import * as stream from 'stream';
+import { Download } from 'src/model/download.entity';
+import { Version } from 'src/model/version.entity';
+
+@Controller('api/v1/packages')
+export class PackagesController {
+  constructor(private readonly em: EntityManager) {}
+
+  @Get(':name')
+  async getPackageName(@Param('name') name: string, @Param('version') version: string) {
+
+    const pkg = await this.em.findOne(Package, { name }, { populate: ['versions'] });
+    if (!pkg) {
+      throw new NotFoundException(`Package "${name}" not found`);
+    }
+
+    const ver = pkg.versions.getItems();
+
+    if (!ver) {
+      throw new NotFoundException(`Version "${version}" not found for package "${name}"`);
+    }
+
+    return {
+      name: pkg.name,
+      versions: ver,
+      readme: pkg.readme,
+      tag_name: pkg.tagName,
+    };
+  }
+
+  @Get(':name/:version')
+  async getPackage(@Param('name') name: string, @Param('version') version: string) {
+    const [major, minor, patch] = version.split('.').map(Number);
+
+    const pkg = await this.em.findOne(Package, { name }, { populate: ['versions'] });
+    if (!pkg) {
+      throw new NotFoundException(`Package "${name}" not found`);
+    }
+
+    const ver = pkg.versions.getItems().find(
+      v => v.major === major && v.minor === minor && v.patch === patch,
+    );
+
+    if (!ver) {
+      throw new NotFoundException(`Version "${version}" not found for package "${name}"`);
+    }
+
+    return {
+      name: pkg.name,
+      version: ver.versionNumber,
+      size_kb: ver.sizeKb,
+      readme: pkg.readme,
+      created_at: ver.createdAt,
+      tag_name: pkg.tagName,
+    };
+  }
+
+
+
+  @Get(':name/latest/download')
+  async downloadLatestPackage(
```

Comment:
Let's remove this endpoint. 

Our CLI tool needs information about version it downloads, because it must also update `.toml` file. So if client has version - can use our version specific download endpoint, not the "latest". I thought it can be useful, but no - we don't need it.